### PR TITLE
Allow users to specify a link target

### PR DESCRIPTION
**Pull request by Ardnived**
When you make some text into a link you are given a prompt to set the URL. This adds a second prompt where you can set the `target` attribute of the link.

### CODE REVIEW

File: src/blocks/scribe-plugins/scribe-link-prompt-plugin.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+  // ===== TRANSFORMS ===== //
+  function runTransforms(transforms, initialLink) {
+    return transforms.reduce(function(currentLinkValue, transform) {
+      return transform(currentLinkValue);
+      }, initialLink);
+  }
+
+  return function(scribe) {
+    const linkPromptCommand = new scribe.api.Command('link');
```
Comment (raffij) [inline src/blocks/scribe-plugins/scribe-link-prompt-plugin.js:88]: I think it's better to keep the command as `linkPrompt`. If not the toolbar will need to be changed too and a migration path for old installs.

**Comment by Ardnived** [comment]
> Agree this works. The target prompt feels a little cumbersome. Going to have a think about it.

Definitely cumbersome. I was also considering doing it as a Yes / No prompt instead of an input. Which would be a bit better, though not by a lot.

**Comment by Ardnived** [comment]
Hey @raffij, I'm going to resume working on this unless you are doing any work on this.

It seems like the most straightforward solution would be to create a basic/lightweight modal that lets you input the link and choose if it should open in a new tab whenever you click the link format button.

Do you have any specific thoughts on this before I go ahead?

**Comment by raffij** [comment]
Yeah either a tool tip that appears or a modal is good. Let's have the link url and a checkbox for open in new window. One small code modal I've seen before is https://micromodal.now.sh/ if that helps with accessibility.

**Comment by Ardnived** [comment]
Implemented a modal using Micromodal.js which takes care of the link itself and opening in a new tab. The other validation still happens in a prompt. I also went through and made the text for this plugin translatable.

Screenshot 2020-05-18 at 15 55 41

**Comment by raffij** [comment]
Looks good. Does it keep the settings when you go back to edit an existing link too? I'll take a proper look later.

**Comment by Ardnived** [comment]
> Looks good. Does it keep the settings when you go back to edit an existing link too? I'll take a proper look later.

It does.

File: package.json [hunk 2]
```diff
@@ -17,7 +17,9 @@
     "verbose": true
   },
   "dependencies": {
+    "@micromodal/repo": "https://github.com/andreyvolokitin/Micromodal.git#patch-1",
```
Comment (raffij) [inline package.json:20]: I should have said we should fork the branch to made by many. A branch might change and include bad code in the future.

Comment (Ardnived) [reply package.json:20]: Good point, I'll do that. I was thinking about that possibility, but didn't think about forking it ourselves.

**Pull request merged**